db/migrations/0005.sql

```diff
@@ -1,9 +1,32 @@
 -- +migrate Down
+DROP VIEW IF EXISTS vw_batch_num_gaps;
 ALTER TABLE data_node.offchain_data DROP COLUMN IF EXISTS batch_num;
 
 -- +migrate Up
 ALTER TABLE data_node.offchain_data
     ADD COLUMN IF NOT EXISTS batch_num BIGINT NOT NULL DEFAULT 0;
 
+-- Ensure batch_num is indexed for optimal performance
+CREATE INDEX idx_batch_num ON data_node.offchain_data(batch_num);
+
+-- Create a view to detect gaps in batch_num
+CREATE VIEW vw_batch_num_gaps AS
```

Comment:
Not sure about using views, should be just a query from the backend, it can make it easier to migrate db engines in the future.

Reply:
Agree, easier to maintain, done.